components/script/dom/htmlmediaelement.rs

```diff
@@ -200,6 +202,10 @@ pub struct HTMLMediaElement {
     /// https://html.spec.whatwg.org/multipage/#dom-media-played
     #[ignore_malloc_size_of = "Rc"]
     played: Rc<DomRefCell<TimeRangesContainer>>,
+    // https://html.spec.whatwg.org/multipage/#dom-media-audiotracks
+    audio_tracks_list: MutNullableDom<AudioTrackList>,
```

Comment:
Not sure whether MutNullableDom is correct here or whether some other area is missing for tracks. This would cause a build failure for attempting to share JSObject between threads when removing the audio and video tracks from the tracks lists in the dedicated media source failure steps. Step 2:

https://html.spec.whatwg.org/multipage/media.html#dedicated-media-source-failure-steps

Reply:
Ahh nevermind, I was referring to here:

https://github.com/servo/servo/blob/bffe2a699e9f0b2c41aa811ea2c3e90e40342419/components/script/dom/htmlmediaelement.rs#L788

but didnt notice the this.root() assignment for 'this' here:

https://github.com/servo/servo/blob/bffe2a699e9f0b2c41aa811ea2c3e90e40342419/components/script/dom/htmlmediaelement.rs#L775